governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java

```diff
@@ -0,0 +1,206 @@
+package com.netflix.governator.lifecycle;
+
+import org.objectweb.asm.*;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.objectweb.asm.Type.*;
+
+public class AnnotationFinder extends ClassVisitor
+{
+    private Set<Type> annotationTypes;
+
+    private Set<Class<?>> annotatedClasses = Collections.emptySet();
+    private Set<Method> annotatedMethods = new HashSet<>();
+    private Set<Constructor> annotatedConstructors = new HashSet<>();
+    private Set<Field> annotatedFields = new HashSet<>();
+
+    private String className;
+    private Class<?> clazz;
+    private ClassLoader classLoader;
+
+    private Class<?> selfClass()
+    {
+        if(clazz == null)
+            clazz = classFromInternalName(className);
+        return clazz;
+    }
+
+    private Class<?> classFromInternalName(String name)
+    {
+        try {
+            return Class.forName(name.replace('/', '.'), true, classLoader);
```

Comment:
In thought that as well, but was following what xbean-finder was doing internally.